### DISCUSSION TITLE
Tempo | Service Graph / Span Metrics | Configurable Metrics Names

### DISCUSSION THREAD

**Discussion by sherifkayad** [grafana/grafana#61653] [Observability]
Currently enabling the Service Graph / Span Metrics in Grafana and while using Tempo as a datasource, it seems that almost all the metrics names used for visualization are hardcoded.

E.g. visualizations related to the Node Graph statistics (RED) or the APM table are depending on some metrics with hardcoded names. For example:
- https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/tempo/QueryEditor/ServiceGraphSection.tsx
- https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/tempo/graphTransform.ts
- ...

I would love to see the names of those metrics given as a configuration to the Tempo datasource.

**Motivation / Reason:**
- Not every one will really use the Metrics Reporter of Tempo .. A good alternative for it (especially in multi-cluster setups) would be the Open Telemtry Collector Service Graph and Span Processor Plugins (these are actually based on the good work of Grafana on the Tempo Metrics Reporter). The OTLP Collector Plugin generates different metrics names that would be really great to have in Grafana visualized for the same purpose like the metrics of the Metrics Processor.
- The metrics names could / would change (either from Tempo or from the OTLP Collector (since the processor plugins are still more or less beta)) .. so it would be cumbersome to change the names in Grafana each time the metrics names change.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82931. Feel free to continue the discussion around this there. Thank you!
